common4j/src/main/com/microsoft/identity/common/java/controllers/BaseController.java

```diff
@@ -862,11 +864,14 @@ protected AccountRecord getCachedAccountRecord(
                     );
         }
 
+        // TO-DO can we remove the parameters.getOAuth2TokenCache() instanceof MsalOAuth2TokenCache condition?
         if (null == targetAccount && parameters.getOAuth2TokenCache() instanceof MsalOAuth2TokenCache) {
             targetAccount = getAccountWithFRTIfAvailable(
                     parameters,
                     (MsalOAuth2TokenCache) parameters.getOAuth2TokenCache()
             );
+        } else if (null == targetAccount && parameters.getOAuth2TokenCache() instanceof BrokerOAuth2TokenCache) {
```

Comment:
I mean implementation here doesn't check that the client is authorized to use the token in the cache if it's found.  I think we have to know whether the client performing the silent request is authorized to share tokens (is a member of the Microsoft Family).  Since the BrokerOAuth2TokenCache relies on the the broker application metadata cache to determine whether or not an application is a member of the Microsoft family... perhaps we should be checking the metadata for the current app client id/application identifier and if not present.... inserting it.

Reply:
Using the same check available in BrokerUtil.getCacheRecordListFromBrokerCache() method to see if the authorized to use the token in the cache if it's found. This change is present in https://github.com/AzureAD/ad-accounts-for-android/pull/1951